packages/tizen_package_manager/tizen/src/tizen_package_manager_plugin.cc

```diff
@@ -218,20 +219,31 @@ class TizenPackageManagerPlugin : public flutter::Plugin {
   }
 
   void GetAllPackagesInfo(std::unique_ptr<FlMethodResult> result) {
-    TizenPackageManager &package_manager = TizenPackageManager::GetInstance();
-    std::optional<std::vector<PackageInfo>> packages =
-        package_manager.GetAllPackagesInfo();
-    if (!packages.has_value()) {
-      result->Error(std::to_string(package_manager.GetLastError()),
-                    package_manager.GetLastErrorString());
-      return;
-    }
-
-    flutter::EncodableList list;
-    for (const PackageInfo &package : packages.value()) {
-      list.push_back(flutter::EncodableValue(PackageInfoToMap(package)));
-    }
-    result->Success(flutter::EncodableValue(list));
+    // TizenPackageManager::GetAllPackagesInfo() is an expensive operation and
+    // might cause unresponsiveness on low-end devices if run on the platform
+    // thread.
+    ecore_thread_run(
+        [](void *data, Ecore_Thread *thread) {
+          auto *result = static_cast<FlMethodResult *>(data);
+
+          TizenPackageManager &package_manager =
+              TizenPackageManager::GetInstance();
+          std::optional<std::vector<PackageInfo>> packages =
+              package_manager.GetAllPackagesInfo();
+          if (packages.has_value()) {
+            flutter::EncodableList list;
+            for (const PackageInfo &package : packages.value()) {
+              list.push_back(
+                  flutter::EncodableValue(PackageInfoToMap(package)));
+            }
+            result->Success(flutter::EncodableValue(list));
```

Comment:
Is it safe calling `Success`(or `Error`) in worker thread?  I think it's safe to return to the main thread and reply back.
(main thread.  Why don't you consider using the second and third arguments of `ecore_thread_run`?

Reply:
If there's no problem with this patch , I'm also fine :)
However, I think it is necessary to add handling in case the start of the worker thread fails or is cancelled.
so  I think you should delete pointer of `result`(passed as user data) in `func_cancel`.

Reply:
The thread is never canceled, so the only case we need to consider is the thread failing to start (according to the documentation). I'll update the implementation. Thanks!